api/envoy/type/matcher/v3/http_inputs.proto

```diff
@@ -60,3 +60,17 @@ message HttpResponseTrailerMatchInput {
   string header_name = 1
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME strict: false}];
 }
+
+// Match input indicates that matching should be done on a specific cookie key.
```

Comment:
Please add a release note for this new feature. @kyessenov what do we need to do to get this matcher to show up in the docs? I forget where we landed on this in terms of what is an extension category, etc.

Reply:
1. Add the new extension to the list of extensions in `extension_manifest.yaml`. Tag it as implementing the HTTP matcher category.
2. Add a comment to the proto as in `// [#extension: envoy.matching.inputs.cookies]`
3. Reference the extension here: https://github.com/envoyproxy/envoy/blob/main/docs/root/intro/arch_overview/advanced/matching/matching_api.rst#http-input-functions.

I'm working to get (1)-(2) validated once we merge https://github.com/envoyproxy/envoy/pull/21413.

Reply:
Thanks @kyessenov for this! Done in https://github.com/envoyproxy/envoy/pull/21496/commits/3930b3ab6fa75cfc523995bffd031f4f3a93e70f